### PR TITLE
fix: SOF-1444 prevent crashing after 180 camera cuts

### PR DESCRIPTION
**Pull request by ianshade**
Refactors persisted audio not to collect excessively nested data, which caused mongo errors.

### CODE REVIEW

File: src/tv2-common/onTimelineGenerate.ts [hunk 1]
```diff
@@ -295,53 +320,72 @@ export function createSisyfosPersistedLevelsTimelineObject(
...
-		if (doesMetaDataNotAcceptPersistAudioDeep(sisyfosPersistMetaData)) {
-			break
-		}
+	return Array.from(new Set(layersToPersist))
+}
 
-		if (i === sortedPieces.length - 1) {
-			layersToPersist.push(...sisyfosLayersThatWantsToBePersisted)
-		}
+function findPersistedLayers(
```
Comment (LindvedKrvang) [inline src/tv2-common/onTimelineGenerate.ts:352]: This method and `findLayersToPersistOnPartEnd` are really similar. Would guess around 80%. Is it possible to extract some of the duplication away?

**Review by LindvedKrvang — APPROVED** [review]
Some good changes in this.
And it's hard to argue against deleting code ;)

File: src/tv2-common/actions/executeAction.ts [hunk 2]
```diff
@@ -1052,15 +1045,15 @@ async function executePiece<
 	shouldBeQueued: boolean,
 	partToQueue: IBlueprintPart
 ) {
-	const currentPieceInstances = await context.core.getPieceInstances('current')
+	const currentPieceInstances = await context.core.getResolvedPieceInstances('current')
```
[the remark this answers is not in the document]
Comment (ianshade) [reply src/tv2-common/actions/executeAction.ts:1048]: `getResolvedPieceInstances` does some extra work (including resolving the _piece timeline_) to make sure we're getting the pieces that are/were actually active and not pieces occluded by others on the same layer, etc., as a result, the `resolvedStart` it gives us is more trustworthy than my previous attempt at calculating the start time based on what we got from `getPieceInstances`

**Review by KvelaGorrrrnio — APPROVED** [review]
🚀

**Review by RasmusAlbrektsen — APPROVED** [review]

**Pull request merged**